### PR TITLE
Theming/buttons

### PR DESCRIPTION
**Pull request by vyvil**
PR (to Theme branch) to update Button component
(Includes CircleButtons and Spinner as well).

Ir might look like its a big PR, but mostly its deleting files / irrelevant components and no longer used props.

### CODE REVIEW

File: src/components/Spinner/Spinner.js [hunk 1]
```diff
@@ -19,43 +19,42 @@
...
+const Spinner = (props: Props) => {
+  const {
+    size = 40,
+    trackWidth = 3,
+    color,
+    style,
+  } = props;
+  const spinnerColor = getSpinnerColor(props);
+  return (
+    <StyledMaterialIndicator size={size} trackWidth={trackWidth} color={color || spinnerColor} style={style} />
```
Comment (wjkg) [inline src/components/Spinner/Spinner.js:56]: Is the `color || spinnerColor` condition here on purpose? If it were included in `getSpinnerColor`, there would be only one place that decides which color gets applied.

**Review by graszka22 — COMMENTED** [review]
Some spinners has no updated props (width, height => size), for example this one in collapsible section:

Screenshot 2020-10-26 at 11 47 17

Go to migrate assets to smart wallet -> select anything, next -> the confirm button has no centered text:

Screenshot 2020-10-26 at 11 57 50

On a side note, is it just me or this grey button looks like it's disabled? Maybe it's a good idea to discuss that with Dmitry :) 

Screenshot 2020-10-26 at 12 11 42

File: src/components/Button/Button.js [hunk 2]
```diff
@@ -444,138 +251,63 @@ class Button extends React.Component<Props, State> {
...
       rightIconName,
       rightIconStyle,
+      small,
+      textStyle,
     } = this.props;
 
     if (isLoading) {
       return (
-        <Spinner width={20} height={20} />
+        <Spinner size={small ? 16 : 24} trackWidth={small ? 1 : 2} basic style={{ paddingLeft: 8, paddingRight: 8 }} />
```
[the remark this answers is not in the document]
Comment (vyvil) [reply src/components/Button/Button.js]: Not sure - I guess such case is unlikely. I've updated it, though - set it to be darker when it's light mode and secondary button.

**Comment by vyvil** [comment]
Thats for review @graszka22 and @wjkg :)
I've fixed everything.

As for secondary button's color on dark mode.
Fair point. I guess it seems like disabled due to shades' contrast.
Surfaces everywhere should not be blue shaded (I've skipped updating modals' color as they were being updated).
When all components have desaturated colors, those buttons does not look like disabled.
This is how it's gonna look like in the end:

modal

**Review by IAmKio — APPROVED** [review]

**Review by graszka22 — APPROVED** [review]

**Review by wjkg — APPROVED** [review]

**Pull request merged**